### DISCUSSION TITLE
How to correctly dispose whole graph?

### DISCUSSION THREAD

**Discussion by tmarsik42** [antvis/X6#3723] [Q&A]
When I navigate in my app the nodes from different graphs are getting combined in one.
I tried graph.dispose, graph.resetCells, graph.removeCells, view.empty, assigning graph to undefined, even unmounting and mounting the whole graph component doesn't work.

This is the first route with only 2 nodes and one edge
<img width="1322" alt="Snímek obrazovky 2023-06-30 v 14 20 11" src="https://github.com/antvis/X6/assets/95856146/f92292bd-acdd-47c5-a760-1b04868cd8bd">

Then I navigate forward to a page with a more difficult graph (which contains previous 2 nodes too).
<img width="1354" alt="Snímek obrazovky 2023-06-30 v 14 21 54" src="https://github.com/antvis/X6/assets/95856146/5f16caa2-9abf-4a4d-87da-5d2fe3273256">
(you can see the edge that looks like it doesnt belong, it connects nodes from previous page)

This is the same page after hard reload (this is how it's supposed to look like).
<img width="1354" alt="Snímek obrazovky 2023-06-30 v 14 22 36" src="https://github.com/antvis/X6/assets/95856146/c695320f-f059-43f1-9c3e-d5ea1d487db2">

This is the result after navigating backwards from the second page to first page (first image is how it's supposed to look like)
<img width="1349" alt="Snímek obrazovky 2023-06-30 v 14 24 02" src="https://github.com/antvis/X6/assets/95856146/659e2c0f-78e7-4f9d-9542-c3ecce2557bf">
You can see how those original 2 nodes are in a top left corner connected and missing from the big graph (that shouldnt be there anyways).

It's clear that it's somehow leaking nodes between pages which I dont want to. How am I supposed to dispose of it when graph.dispose() doesnt work?

**Comment by NewByVector** [comment]
Can you write a simple demo to rehash your problem and raise an issue?

**Reply by tmarsik42** [reply]
Thanks for a response. I made a repro, it's available here: https://stackblitz.com/edit/sveltejs-kit-template-default-afemn6
I tried to simplify it as much as possible (ommited the whole ELK thing).

Though the reproducing is a bit random and takes some clicking. Just keep navigating between /1 and /2 and refresh until this occurs (diagonal edge on the right side). That is the sign those 2 graphs got merged like on my original screenshots.

<img width="936" alt="Snímek obrazovky 2023-07-04 v 12 56 12" src="https://github.com/antvis/X6/assets/95856146/62c5bbec-cbad-431b-8389-4f82d7e2fd2c">

I don't think it is a bug honestly, I'm probably just doing something wrong. I'll definitely raise an issue though once I'm sure it's not my mistake.

**Reply by tmarsik42** [reply]
After some more digging into this problem I've made a (hopefully) better reproduction and an issue https://github.com/antvis/X6/issues/3751
